### ISSUE TITLE
Minimized report causes JS error

### ISSUE THREAD

**Issue by mrakitin** [radiasoft/sirepo#300]
Minimized report after maximizing causes this JS error both in the browser console and server log:
```js
10.0.2.2 - - [18/Aug/2016 15:19:22] "POST /run-simulation HTTP/1.1" 200 -
../sirepo/server.py:199:app_error_logging 10.0.2.2: javascript error: {
    "cause": "",
    "message": "Error: [$rootScope:inprog] http://errors.angularjs.org/1.4.2/$rootScope/inprog?p0=%24digest",
    "stackTrace": [
        "J/<@http://localhost:8000/static/js/angular-1.4.2.min.js:6:416",
        "r@http://localhost:8000/static/js/angular-1.4.2.min.js:127:187",
        "jf/this.$get</n.prototype.$digest@http://localhost:8000/static/js/angular-1.4.2.min.js:131:374",
        ".linkPlot/</scope.windowResize<@http://localhost:8000/static/js/sirepo-plotting.js?20160818.151851:179:21",
        "debounce/</later@http://localhost:8000/static/js/sirepo-plotting.js?20160818.151851:26:17",
        "td/g.prototype.notify/<@http://localhost:8000/static/js/angular-1.4.2.min.js:123:1",
        "jf/this.$get</n.prototype.$eval@http://localhost:8000/static/js/angular-1.4.2.min.js:134:489",
        "jf/this.$get</n.prototype.$digest@http://localhost:8000/static/js/angular-1.4.2.min.js:132:1",
        "jf/this.$get</n.prototype.$apply@http://localhost:8000/static/js/angular-1.4.2.min.js:135:267",
        "e/p.$$intervalId<@http://localhost:8000/static/js/angular-1.4.2.min.js:96:315"
    ],
    "type": "exception",
    "url": "http://localhost:8000/srw#/beamline/3iRUqjkg?application_mode=wavefront"
}
10.0.2.2 - - [18/Aug/2016 15:19:23] "POST /error-logging HTTP/1.1" 200 -
```
It can be reproduced in any example, but I used `Young's Double Slit Experiment` example and minimized `Intensity at W30 Report, 30m` report, changed some parameters (e.g., sampling factor) to trigger recalculation of the reports and then maximized the minimized report.

**Comment by mrakitin** [comment]
It even does not need recalculation, you just need to minimize/maximize a report.

**Comment by robnagler** [comment]
I don't understand this change;

https://github.com/radiasoft/sirepo/commit/31f9c5a159137ebb1e58bc5f4a65da617c99e3f4#diff-bd59f22ea95f9de04f76e1fe585c1e14R176

The $scope.apply was unnecessary so I removed it. I don't think you want to
run $digest explicitly.

**Comment by moellep** [comment]
The $apply or $digest is needed or the reports don't resize with the page.
You need to run $digest because the event comes in from outside angular. If
it also gets called from within angular, you get the "digest in progress"
error. I'll debug it.

[image: Inline image 1]

**Comment by robnagler** [comment]
The event comes from angular, because I changed it to $interval.

**Comment by moellep** [comment]
I reverted the debounce() back to setTimeout() / clearTimeout(). The change
to $interval also caused the callback to get lost for everything except the
last registered caller.

Paul

**Comment by robnagler** [comment]
We can't use setTimeout because of
https://github.com/angular/protractor/issues/169

**Comment by robnagler** [comment]
How is it supposed to work? I thought the function should only be called once no matter how many times debounce was called.

**Comment by moellep** [comment]
Each plot registers to get called back on a resize event. When the resize
happens, the windowResize gets called for that scope:

scope.windowResize = debounce(function() {
    console.log('resize: ', scope.modelName);
    scope.resize();
    scope.$digest();
}, 250);

For some reason, the $interval code was only calling windowResize() for the
plot which registerred last, not all the plots.

Paul

**Comment by robnagler** [comment]
I understand. I'll fix

**Comment by robnagler** [comment]
Fix 8e7b244
